Library/Homebrew/utils.rb

```diff
@@ -188,14 +188,32 @@ def self.homebrew_version_string
 
   def self.install_gem_setup_path!(gem, version = nil, executable = gem)
     require "rubygems"
-    ENV["PATH"] = "#{Gem.user_dir}/bin:#{ENV["PATH"]}"
 
-    args = [gem]
-    args << "-v" << version if version
+    if Gem::Specification.find_all_by_name(gem, version).empty?
+      ohai "Installing missing or updating outdated '#{gem}' gem"
```

Comment:
My primary reason still stands (output before stalling with network action). But I have to admit that the verbosity in my example is due to me having moved `ENV["PATH"] =` (now back to the top) and mostly due to `bundler` (invoked by `brew tests`). Just in case you'd like to reconsider.

More representative output with profiling information is forthcoming.